ApolloSQLite/SQLiteNormalizedCache.swift

```diff
@@ -83,60 +83,56 @@ public final class SQLiteNormalizedCache: NormalizedCache {
       throw SQLiteNormalizedCacheError.invalidRecordEncoding(record: record)
     }
 
-    let recordJSON = try SQLiteSerialization.deserialize(data: recordData)
-    return Record(key: row[key], recordJSON)
+    let fields = try SQLiteSerialization.deserialize(data: recordData)
+    return Record(key: row[key], fields)
   }
 }
 
 private let serializedReferenceKey = "reference"
 
 final class SQLiteSerialization {
-  static func serialize(fields: JSONObject) throws -> Data {
+  static func serialize(fields: Record.Fields) throws -> Data {
     var objectToSerialize = JSONObject()
     for (key, value) in fields {
       objectToSerialize[key] = try serialize(value: value)
     }
-    return try JSONSerializationFormat.serialize(value: objectToSerialize)
+    return try JSONSerialization.data(withJSONObject: objectToSerialize, options: [])
+  }
+  
+  private static func serialize(value: Record.Value) throws -> JSONValue {
+    switch value {
+    case let reference as Reference:
+      return [serializedReferenceKey: reference.key]
+    case let array as [Record.Value]:
+      return try array.map { try serialize(value: $0) }
+    default:
+      return value
+    }
   }
 
-  static func deserialize(data: Data) throws -> JSONObject {
-    let object = try JSONSerializationFormat.deserialize(data: data)
+  static func deserialize(data: Data) throws -> Record.Fields {
+    let object = try JSONSerialization.jsonObject(with: data, options: [])
     guard let jsonObject = object as? JSONObject else {
       throw SQLiteNormalizedCacheError.invalidRecordShape(object: object)
     }
-    var deserializedObject = JSONObject()
+    var fields = Record.Fields()
     for (key, value) in jsonObject {
-      deserializedObject[key] = try deserialize(valueJSON: value)
+      fields[key] = try deserialize(jsonValue: value)
     }
-    return deserializedObject
+    return fields
   }
 
-  private static func deserialize(valueJSON: Any) throws -> Any {
-    switch valueJSON {
+  private static func deserialize(jsonValue: JSONValue) throws -> Record.Value {
```

Comment:
How about this for the naming of the `serialize` and `deserialize` method parameters, for even greater clarity?
- `serialize(fieldValue:)`
- `serialize(fieldJSON:)`